### ISSUE TITLE
EFCore doesn't add ORDER BY clause (OrderBy_nav_prop_reference_optional fails)

### ISSUE THREAD

**Issue by roji** [aspnet/EntityFramework#5215]
While trying to run tests on Npgsql, the OrderBy_nav_prop_reference_optional test fails. Debugging this shows that the following test code produces SQL without an ORDER BY clause:

```c#
expected = context.LevelOne
    .Include(e => e.OneToOne_Optional_FK)
    .ToList()
    .OrderBy(e => e?.OneToOne_Optional_FK?.Name)
    .Select(e => e.Id)
    .ToList();
```

Under SqlServer the test happens to pass because the expected order is returned by chance.

**Comment by rowanmiller** [comment]
This particular code snippet intentionally does the ordering client side (and compares to a second wuery with ordering done in db). Assigning to investigate though.

**Comment by maumar** [comment]
fixed in d344ec0025b7ca37604d6a65a000234093ccb619
